src/typ/expr.rs

```diff
@@ -69,17 +78,17 @@ impl<'a, 'b, 'c> ExprTyper<'a, 'b, 'c> {
     ///
     pub fn infer(&mut self, expr: UntypedExpr) -> Result<TypedExpr, Error> {
         match expr {
-            UntypedExpr::ListNil { location, .. } => self.infer_nil(location),
+            UntypedExpr::ListNil { location, .. } => Ok(self.infer_nil(location)),
```

Comment:
If these functions don't return a result they should not be called infer_ any more. `make_nil` perhaps. Same for the others.

Reply:
Okay, I'll work on it.